AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java

```diff
@@ -82,6 +86,17 @@
     private Runnable initialVideoToggleRequest;
     private AudioDeviceSelectionPopupWindow audioDeviceSelectionPopupWindow;
     private ParticipantListPopupWindow participantListPopupWindow;
+    private ImageButton deviceOptionsButton;
+
+    private final Consumer<AudioDeviceType> audioDeviceTypeConsumer = new Consumer<AudioDeviceType>() {
+        @Override
+        public void accept(final AudioDeviceType audioDeviceType) {
+            final Drawable audioDevice = ContextCompat.getDrawable(CallActivity.this,
+                    audioDeviceType == AudioDeviceType.ANDROID
+                            ? R.drawable.ic_fluent_speaker_2_28_filled : R.drawable.ic_fluent_speaker_2_28_regular);
+            deviceOptionsButton.setImageDrawable(audioDevice);
```

Comment:
Does this get run in the UI thread?

Reply:
Yes, confirmed with debugger that `audioDeviceTypeConsumer.accept()` is run in ActivityThread.java, which is the Main thread loop